File: programmers/코딩테스트 고득점 Kit/스택, 큐/기능개발.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
int chk[101];
vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	int Max = 0;
	for (int i = 0; i < progresses.size(); i++) {
		int com = progresses[i];
		int spd = speeds[i];
		for (int j = 1; j < 100; j++) {
			if (com + j * spd >= 100) {
				Max = max(Max, j);
				chk[Max]++;
				break;
			}
		}
	}
	for (int i = 1; i < 100; i++)
		if (chk[i]) answer.push_back(chk[i]);
	return answer;
}

File: programmers/코딩테스트 고득점 Kit/스택, 큐/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int chk[] = new int[100];
        int afterDays = 0;
        for(int i=0;i<progresses.length;i++) {
            for(int day=1;day<100;day++) {
                if(progresses[i] + day * speeds[i] >= 100) {
                    afterDays = Math.max(afterDays, day);
                    chk[afterDays]++;
                    break;
                }
            }
        }
        List<Integer> list = new ArrayList<>();
        for(int i=1;i<100;i++) if(chk[i] > 0) list.add(chk[i]);
        int[] answer = new int[list.size()];
        for(int i=0;i<answer.length;i++) answer[i] = list.get(i);
        return answer;
    }
}